lib/daru.rb

```diff
@@ -65,11 +65,18 @@ def create_has_library(library)
   create_has_library :nyaplot
 end
 
-autoload :Spreadsheet, 'spreadsheet'
-autoload :CSV, 'csv'
+[['reportbuilder', '~>1.4'], ['spreadsheet', '~>1.1.1']].each do |lib|
+  begin
+    gem lib[0], lib[1]
+    require lib[0]
+  rescue LoadError
+    STDERR.puts "\nInstall the #{lib[0]} gem version #{lib[1]} for using"\
```

Comment:
That's a good idea, but will involve shifting the `require` to when the method is being called, which can increase the time required to execute the method.

However, I think we can achieve something like what you're saying if daru can be modularized effectively as per this proposal: https://github.com/SciRuby/daru/pull/304